packages/dtslint/src/index.ts

```diff
@@ -265,16 +269,19 @@ async function testTypesVersion(
   expectOnly: boolean,
   tsLocal: string | undefined,
   isLatest: boolean,
-): Promise<void> {
-  assertIndexdts(dirPath);
+): Promise<{ errors: string[] }> {
+  const errors = [];
+  const checkExpectedFilesResult = checkExpectedFiles(dirPath, isLatest);
+  errors.push(...checkExpectedFilesResult.errors);
   const tsconfigErrors = checkTsconfig(dirPath, getCompilerOptions(dirPath));
   if (tsconfigErrors.length > 0) {
-    throw new Error("\n\t* " + tsconfigErrors.join("\n\t* "));
+    errors.push("\n\t* " + tsconfigErrors.join("\n\t* "));
```

Comment:
It may soon be worth collecting structured error data and making a nice reporter/formatter, or using a dependency if you know of a good one. Our newlines and indentation are pretty inconsistent.

Reply:
Yeah, I saw this and was like "Oh yeah this has some pretty formatting, doesn't it".

From what I can tell, this eventually gets thrown as an error, so as long as it adds a leading newline, it actually turns out okay. But we don't do that in many cases.